R/fit.R

```diff
@@ -51,23 +55,27 @@
 #'   add_recipe(recipe)
 #'
 #' fit(recipe_wf, mtcars)
-fit.workflow <- function(object, data, ..., control = control_workflow()) {
+fit.workflow <- function(object, data, ..., calibration = NULL, control = control_workflow()) {
   check_dots_empty()
 
   if (is_missing(data)) {
     cli_abort("{.arg data} must be provided to fit a workflow.")
   }
 
+  validate_has_calibration(object, calibration)
+
   if (is_sparse_matrix(data)) {
     data <- sparsevctrs::coerce_to_sparse_tibble(data)
   }
 
   workflow <- object
   workflow <- .fit_pre(workflow, data)
   workflow <- .fit_model(workflow, control)
-  # if (has_postprocessor(workflow)) {
-  #   workflow <- .fit_post(workflow, calibration)
-  # }
+  if (has_postprocessor(workflow)) {
+    # if (is.null(calibration)), then the tailor doesn't have a calibrator
+    # and training the tailor on `data` will not leak data
+    workflow <- .fit_post(workflow, calibration %||% data)
```

Comment:
Maybe:

```r
  if (has_postprocessor(workflow)) {
    # if (is.null(calibration)), then the tailor doesn't have a calibrator
    # and training the tailor on `data` will not leak data
    workflow <- .fit_post(workflow, calibration %||% data)
  }
```

Could become:

```r
if (!.workflow_needs_calibration_data(workflow)) {
  calibration <- data
}

if (has_postprocessor(workflow)) {
  workflow <- .fit_post(workflow, data)
}
```

?

Maybe still worth including a comment there in the latter.

Reply:
I think this here is actually a good argument for not calling the function `.workflow_needs_calibration_data()` because it might be hella confusing on the first read when we assign something to `calibration` right after checking that the workflow doesn't need that 😆 😅

Reply:
b28a6c4!

Reply:
> I think this here is actually a good argument for not calling the function .workflow_needs_calibration_data() because it might be hella confusing on the first read when we assign something to calibration right after checking that the workflow doesn't need that 😆 😅

HAHAHAHA